### PR TITLE
Remove duplicate lines

### LINKED ISSUE #2: propagate `pytree_dataclass` typechecking through vectorizing transformations
A rough sketch of the way a transformer is implemented in `train.py` might be 
```python
@pytree_dataclass
class Model:
  w_up: f32['layers d_model/d d_ff/t']
  # ...
  
@typechecked
def transformer_block(x: f32[b'batch/d seq d_model/t'], w_block: Any) -> f32[b'batch/d seq d_model/t']:
  # ...
  return x, ()
  
@typed_shard_map
def transformer(x: f32[b'batch/d seq d_model/t'], w_model: Model) -> f32[b'batch/d seq d_model/t']:
  y, _ = jax.lax.scan(transformer_block, x, w)
  return y
```
It would be helpful if there was some way of checking that `w_block` had the expected type, i.e the same as `Model` but with the leading `'layers'` axis removed from each field. One very hacky way of doing this is to use
```python
@pytree_dataclass
class Block:
  w_up: f32['d_model/d d_ff/t']
  # ...
  
@pytree_dataclass
class Model(Block):
  w_up: f32['layers d_model/d d_ff/t']
  # ...
  
@typechecked
def transformer_block(x: f32[b'batch/d seq d_model/t'], w_block: Block) -> f32[b'batch/d seq d_model/t']:
# ...
```
However, in more complex situations, this quickly leads to a large amount of boilerplate dataclasses. Ideally, there might be some kind of api that would allow e.g
```python
Model = expand_named_axes('layers', Block)
```
If the maintainers think this is a good idea, I'm happy to start working on a draft implementation.

### LINKED ISSUE #5: nit: duplicate lines in train.pu
as titled, the init of w_kv seems to be duplicated in train.py 

https://github.com/MatX-inc/seqax/blob/3d8aba27487edb1655fcfbfb873fe09e4b67458b/train.py#L99-L100

### PR DESCRIPTION
**Pull request by vcvicek**
Fixes #5

### CODE REVIEW

**Review by tensorpro — APPROVED** [review]
Thanks!

**Review by blackgnezdo — APPROVED** [review]
